Maria-Edgeworth-Letters-Project/me-tei
# Add testing / validation

## What does this Pull Request do?

This PR adds tests that check to see if our XML files are valid and follow the TEI schema.

## How should this be tested?

You can test the tests by doing the following (these instructions are added to the README):

Tests can be run locally by doing the following:

Install shunit2.  On MacOs this can be done with:

```
brew install shunit2
```

You can check if you already have it installed by doing this from the command line:

```
which shunit2
```

Then, run tests by doing:

```
cd tests && ./validate_records.sh
```

## Additional Notes

Here are the invalid records I got when testing - 
[EdgeworthValidationErrors.txt](https://github.com/user-attachments/files/17167012/EdgeworthValidationErrors.txt). After this is merged I plan to add a separate folder where students can add TEI files initially. You and I can periodically (weekly?) open up PRs and run tests against the records in that separate folder. Tests take a while to run, so siphoning them off will make our work easier.
